build: Manage mspdbsrv instances during Win builds

peter-b opened the pull request:
Builds were stalling and timing out on Windows, and this appeared
to be due to `mspdbsrv` instances hanging around after the build
had been completed.

Further investigation revealed a host of issues related to
`mspdbsrv` process management.  This patch updates the Windows
buildbot build script to prevent as many of these issues as
possible.

peter-b committed:
build: Manage mspdbsrv instances during Win builds

Builds were stalling and timing out on Windows, and this appeared
to be due to `mspdbsrv` instances hanging around after the build
had been completed.

Further investigation revealed a host of issues related to
`mspdbsrv` process management.  This patch updates the Windows
buildbot build script to prevent as many of these issues as
possible.

runrevmark commented on buildbot.py:
```diff
@@ -118,15 +123,69 @@ def exec_make(target):
+#   even if a Python exception causes a non-local exit.
+class UniqueMspdbsrv(object):
+    def __enter__(self):
+        os.environ['_MSPDBSRV_ENDPOINT_'] = '{}-{}-{}'.format(
```
Would adding a -<uuid> to this string help ensure complete uniqueness? Just in case the mspdbsrv does not quit from a previous identical run, it would mean a subsequent identical build would use a distinct instance.

peter-b replied:
I guess I could just use a random string.  Correctly generating random strings is a bit of a pain in Python 2.7.  I'll tweak this.

livecodepanos commented:
@livecode-vulcan review ok ae5a76c

livecode-vulcan commented:
💙 review by @livecodepanos ok ae5a76c2fd6f1303e39031a09a67104bbf369116

peter-b committed:
build: Use random UUID as mspdbsrv endpoint

Prevent any chance of `mspdbsrv` endpoint collisions between
consecutive builds of the same commit.

peter-b commented:
@livecodepanos @runrevmark I've tweaked this to use a randomly-generated UUID as the `mspdbsrv` endpoint rather than a constructed string based on the git commit and build metadata.

livecodepanos commented:
@livecode-vulcan review ok 9eac38f

livecode-vulcan commented:
💙 review by @livecodepanos ok 9eac38fd38337e82f527a9cb7b91b972bcd341a8

livecode-vulcan commented:
😎 test success 9eac38fd38337e82f527a9cb7b91b972bcd341a8

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/957)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/963)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/955)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/956)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/963)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/972)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/954)
